Sources/Testing/Traits/Trait.swift

```diff
@@ -12,75 +12,80 @@
+  /// their traits, and before it begins to run any tests.
+  /// Use this method to prepare necessary internal state, or to determine
   /// whether the test should run.
   ///
   /// The default implementation of this method does nothing.
   func prepare(for test: Test) async throws
 
-  /// The user-provided comments for this trait, if any.
+  /// The user-provided comments for this trait.
   ///
-  /// By default, the value of this property is an empty array.
+  /// The default value of this property is an empty array.
   var comments: [Comment] { get }
 
   /// The type of the test scope provider for this trait.
   ///
-  /// The default type is `Never`, which cannot be instantiated. The
-  /// ``scopeProvider(for:testCase:)-cjmg`` method for any trait with this
-  /// default type must return `nil`, meaning that trait will not provide a
-  /// custom scope for the tests it's applied to.
+  /// The default type is `Never`, which can't be instantiated. The
+  /// ``scopeProvider(for:testCase:)-cjmg`` method for any trait with
+  /// `Never` as its test scope provider type must return `nil`, meaning that
+  /// the trait doesn't provide a custom scope for tests it's applied to.
   associatedtype TestScopeProvider: TestScoping = Never
 
-  /// Get this trait's scope provider for the specified test and/or test case,
-  /// if any.
+  /// Get this trait's scope provider for the specified test, and optional test
+  /// case.
   ///
   /// - Parameters:
   ///   - test: The test for which a scope provider is being requested.
   ///   - testCase: The test case for which a scope provider is being requested,
   ///     if any. When `test` represents a suite, the value of this argument is
   ///     `nil`.
   ///
-  /// - Returns: A value conforming to ``Trait/TestScopeProvider`` which may be
-  ///   used to provide custom scoping for `test` and/or `testCase`, or `nil` if
-  ///   they should not have any custom scope.
+  /// - Returns: A value conforming to ``Trait/TestScopeProvider`` which you
+  ///   use to provide custom scoping for `test` and/or `testCase`, or `nil` if
+  ///   the trait doesn't provide any custom scope for the test or test case.
   ///
   /// If this trait's type conforms to ``TestScoping``, the default value
   /// returned by this method depends on `test` and/or `testCase`:
   ///
-  /// - If `test` represents a suite, this trait must conform to ``SuiteTrait``.
+  /// - If `test` represents a suite, this trait also conforms to
+  ///   ``SuiteTrait``.
   ///   If the value of this suite trait's ``SuiteTrait/isRecursive`` property
-  ///   is `true`, then this method returns `nil`; otherwise, it returns `self`.
-  ///   This means that by default, a suite trait will _either_ provide its
-  ///   custom scope once for the entire suite, or once per-test function it
-  ///   contains.
+  ///   is `true`, then this method returns `nil`, and the suite trait
```

Comment:
This discussion only applies to suite traits (it's saying "if the suite is recursive, then this function is recursively called for tests in the suite.")